### PR TITLE
Test for multiple particle interpolation

### PR DESCRIPTION
**Pull request by gassmoeller**
Follow-up to #6052 that adds additional safety checks for interpolation from multiple particle worlds, and tests that we can correctly interpolate particle properties from multiple particle worlds into compositional fields.

### CODE REVIEW

**Comment by gassmoeller** [comment]
Related to #5877.

**Comment by tjhei** [comment]
Would you mind rebasing?

**Comment by gassmoeller** [comment]
Rebased.

File: source/simulator/initial_conditions.cc [hunk 1]
```diff
@@ -326,12 +335,18 @@ namespace aspect
...
 
+                advection_field_has_been_found[advection_field] = true;
                 particle_property_indices[world_index].push_back({advection_field,particle_property_index});
                 property_mask[world_index].set(particle_property_index,true);
               }
           }
       }
 
+    for (unsigned int advection_field=0; advection_field<advection_fields.size(); ++advection_field)
+      Assert (advection_field_has_been_found[advection_field] == true,
```
Comment (tjhei) [inline source/simulator/initial_conditions.cc:346]: would you mind adding ``advection_field`` to the error message?

File: source/simulator/initial_conditions.cc [hunk 2]
```diff
@@ -306,9 +311,13 @@ namespace aspect
                 // If not: assume we find it in another world.
                 if (particle_property_manager.get_data_info().fieldname_exists(particle_property_and_component.first))
                   {
+                    Assert (advection_field_has_been_found[advection_field] == false,
+                            ExcMessage("The same advection field is mapped to particle properties in different particle worlds. This is not supported."));
```
Comment (tjhei) [inline source/simulator/initial_conditions.cc:315]: same here

**Review by tjhei — APPROVED** [review]

**Comment by gassmoeller** [comment]
Something like this?

**Review by tjhei — APPROVED** [review]

**Pull request merged**